Northstar.CustomServers/mod/scripts/vscripts/evac/_evac.gnut

```diff
@@ -422,6 +422,39 @@ void function Evac( int evacTeam, float initialWait, float arrivalTime, float wa
 		foreach ( entity otherPlayer in GetPlayerArray() )
 			Remote_CallFunction_NonReplay( otherPlayer, "ServerCallback_EvacObit", player.GetEncodedEHandle() )
 	}
+
+	// award player score to evacing team
+	int evacCount = 0
+	entity lastEvacPlayer
+	foreach ( entity player in GetPlayerArrayOfTeam( dropship.GetTeam() ) )
+	{
+		if ( !PlayerInDropship( player, dropship ) )
+			continue
+		
+		evacCount++
+		// for sole survivor things
+		lastEvacPlayer = player
+	}
+
+	bool allEvac = evacCount == GetPlayerArrayOfTeam( dropship.GetTeam() ).len()
+
+	foreach(entity player in GetPlayerArrayOfTeam(dropship.GetTeam()))
+	{
+		if ( !PlayerInDropship( player, dropship ) )
+			continue
+
+		AddPlayerScore( player, "HotZoneExtract" )
+		UpdatePlayerStat( player, "misc_stats", "evacsSurvived" )
+
+		if ( allEvac )
+			AddPlayerScore( player, "TeamBonusFullEvac" )
+	}
+
+	// sole survivor (but not the only one on the team)
+	if ( evacCount == 1 && !allEvac )
+	{
+		AddPlayerScore( lastEvacPlayer, "SoleSurvivor" )
+	}
```

Comment:
```suggestion
	// award player score to evacing team
	int evacCount = 0
	array<entity> evacingPlayers = GetPlayerArrayOfTeam( dropship.GetTeam() ) // all players that are supposed to evac in the dropship

	// count how many players are in the dropship
	foreach ( entity player in evacingPlayers )
	{
		if ( !PlayerInDropship( player, dropship ) )
			continue
		
		evacCount++
	}

	bool allEvac = evacCount == evacingPlayers.len()

	foreach(entity player in evacingPlayers)
	{
		if ( !PlayerInDropship( player, dropship ) )
			continue

		AddPlayerScore( player, "HotZoneExtract" )
		UpdatePlayerStat( player, "misc_stats", "evacsSurvived" )

		if ( allEvac )
			AddPlayerScore( player, "TeamBonusFullEvac" )
	}

	// sole survivor (but not the only one on the team)
	if ( evacCount == 1 && !allEvac )
	{
		// we can assume there is one player in the array because otherwise evacCount wouldn't be 1
		AddPlayerScore( evacingPlayers[0], "SoleSurvivor" )
	}
```
small optimization